apps/etterlatte-behandling/src/main/kotlin/behandling/klienter/AxsysKlient.kt

```diff
@@ -0,0 +1,69 @@
+package no.nav.etterlatte.behandling.klienter
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.ClientRequestException
+import io.ktor.client.request.accept
+import io.ktor.client.request.get
+import io.ktor.client.request.header
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.http.isSuccess
+import no.nav.etterlatte.behandling.domain.SaksbehandlerEnhet
+import no.nav.etterlatte.libs.common.logging.NAV_CONSUMER_ID
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.time.Duration
+
+interface AxsysKlient {
+    suspend fun hentEnheterForIdent(ident: String): List<SaksbehandlerEnhet>
+}
+
+class AxsysKlientImpl(private val client: HttpClient, private val url: String) : AxsysKlient {
+    private val logger: Logger = LoggerFactory.getLogger(AxsysKlientImpl::class.java)
+
+    private val enhetCache =
+        Caffeine.newBuilder()
+            .expireAfterWrite(Duration.ofMinutes(15))
+            .build<String, List<SaksbehandlerEnhet>>()
+
+    override suspend fun hentEnheterForIdent(ident: String): List<SaksbehandlerEnhet> {
+        logger.info("Henter enheter fra Axsys for saksbehandlerident $ident.")
+
+        val cachedEnhet = enhetCache.getIfPresent(ident)
+        if (cachedEnhet != null) {
+            return cachedEnhet
+        }
+        return try {
+            val response =
+                client.get("$url/api/v2/tilgang/$ident?inkluderAlleEnheter=false") {
+                    header(NAV_CONSUMER_ID, "etterlatte-behandling")
+                    accept(ContentType.Application.Json)
+                    contentType(ContentType.Application.Json)
+                }
+
+            if (response.status.isSuccess()) {
```

Comment:
Denne fungerer bare dersom `expectSuccess=true` i http-klienten, noe den ikke er. Ser vi har flere av disse rundt i koden så det må nok ryddes opp flere steder.

Reply:
Den forstår jeg ikke helt? 
`public fun HttpStatusCode.isSuccess(): Boolean = value in (200 until 300)`

Reply:
Dårlig formulert fra min side. Og mente i tillegg `expectSuccess=false`. 
`.isSuccess()` vil sånn sett fungere her, _men_ det er ingen vits å sjekke det siden alt som er utenfor `200 ... 300` vil kaste exception. Så det å ha en if på om den er suksess har liten hensikt

Reply:
Fixed